src/shared/firewall-util-iptables.c

```diff
@@ -20,15 +21,45 @@
 #include <linux/netfilter/xt_addrtype.h>
 #include <libiptc/libiptc.h>
 
+#ifndef IF_NAMSIZE
+#define IF_NAMESIZE IFNAMSIZ
+#endif
 #include "alloc-util.h"
 #include "firewall-util.h"
 #include "firewall-util-private.h"
 #include "in-addr-util.h"
 #include "macro.h"
+#include "process-util.h"
 #include "socket-util.h"
 
 DEFINE_TRIVIAL_CLEANUP_FUNC(struct xtc_handle*, iptc_free);
 
+int fw_iptables_init(void) {
+        int r = access("/proc/net/ip_tables_names", F_OK);
+        pid_t pid;
+
+        if (r == 0)
+              return 0;
+
+        r = safe_fork("(modprobe)", FORK_RESET_SIGNALS|FORK_WAIT, &pid);
+        if (r < 0)
+                return r;
+
+        if (r == 0) {
+                const char *cmdline[] = {
+                       "modprobe",
```

Comment:
uh, please do not use modprobe... use libkmod if anything, but shelling out is ugly

I must say I am not much of a fan of this... most of our services run without CAP_SYS_MODULE, and this would change here...

Reply:
Okay, i've dropped this change.